eZ/Publish/Core/Persistence/Legacy/Tests/Content/Location/Gateway/DoctrineDatabaseTest.php

```diff
@@ -27,6 +29,24 @@ protected function getLocationGateway()
         );
     }
 
+    /**
+     * Returns a ready to test DoctrineDatabase gateway.
+     *
+     * @return \eZ\Publish\Core\Persistence\Legacy\Content\Gateway\DoctrineDatabase
+     */
+    protected function getContentGateway()
+    {
+        $databaseGateway = new \eZ\Publish\Core\Persistence\Legacy\Content\Gateway\DoctrineDatabase(
+            ($dbHandler = $this->getDatabaseHandler()),
```

Comment:
The problem with this is that those tests are executed for SQLite only, so they test query integrity only, but don't provide integration testing against any supported platforms.
The test that I asked for should be in the integration namespace, e.g similar to `\eZ\Publish\API\Repository\Tests\LocationServiceTest::testMoveSubtree` (can be right after those existing tests for moving a subtree , e.g. `testMoveSubtreeKeepsContentHidden`).

Reply:
Actually, I have a problem asserting that things are correct in the database using the API.
With the API I cannot check why a object is hidden ( is it hidden on location or content level). This is made transparent by the [domainMapper](https://github.com/ezsystems/ezpublish-kernel/blob/7.5/eZ/Publish/Core/Repository/Helper/DomainMapper.php#L562)
I see no way of actually checking the value of `ezcontentobject_tree.is_hidden` using the API

Is it okay to do SQLs in the integration tests? any other suggestions ? @alongosz

Reply:
> Actually, I have a problem asserting that things are correct in the database using the API. With the API I cannot check why a object is hidden ( is it hidden on location or content level). This is made transparent by the [domainMapper](https://github.com/ezsystems/ezpublish-kernel/blob/7.5/eZ/Publish/Core/Repository/Helper/DomainMapper.php#L562) I see no way of actually checking the value of `ezcontentobject_tree.is_hidden` using the API
> 
> Is it okay to do SQLs in the integration tests? any other suggestions ? @alongosz

Sometimes, but I think what you did is enough, explained in https://github.com/ezsystems/ezpublish-kernel/pull/3142#discussion_r825521987.

Reply:
ok. marking as resolved then

Reply:
Still... this portion of tests does not provide a value as it runs on SQLite only. An integration test is enough.